search/search_engine_base.py

```diff
@@ -7,45 +7,58 @@
 
 
 class SearchEngine:
-
-    """ Base abstract SearchEngine object """
+    """
+    Base abstract SearchEngine object.
+    """
 
     index_name = "courseware"
 
     def __init__(self, index=None):
         if index:
             self.index_name = index
 
-    def index(self, doc_type, sources, **kwargs):
-        """ This operation is called to add documents of given type to the search index """
+    def index(self, sources, **kwargs):
+        """
+        Add documents to the search index.
+        """
         raise NotImplementedError
 
-    def remove(self, doc_type, doc_ids, **kwargs):
-        """ This operation is called to remove documents of given type from the search index """
+    def remove(self, doc_ids, **kwargs):
+        """
+        Remove documents by ids from the search index.
+        """
         raise NotImplementedError
 
-    def search(self,
-               query_string=None,
-               field_dictionary=None,
-               filter_dictionary=None,
-               exclude_dictionary=None,
-               facet_terms=None,
-               **kwargs):  # pylint: disable=too-many-arguments
-        """ This operation is called to search for matching documents within the search index """
+    def search(
+            self,
```

Comment:
I can leave them like they were before, or leave like it is, because code quality checkers are passing. If I improve an indentation checkers are failed